src/dev/ci_setup/setup.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -e
+
+
+###
+### determine directory of kibana project
+###
+dir="$(pwd)"
+if [ -f "$dir/package.json" ] && [ -f "$dir/.node-version" ]; then
+  echo "Setting up node.js and npm in $dir"
+else
+  echo "src/dev/ci_setup/setup.sh must be run within a kibana repo"
+  exit 1
+fi
+
+
+###
+### download node
+###
+nodeVersion="$(cat $dir/.node-version)"
+nodeUrl="https://nodejs.org/download/release/v$nodeVersion/node-v$nodeVersion-linux-x64.tar.gz"
+nodeDir="${NODE_DIR:-"$dir/.node"}/$nodeVersion"
```

Comment:
Can we use a more general cache directory? Possibly as an argument to this script.
This way we could also use it for yarn.

Reply:
If one is not provided, should we default to using `/tmp`?

Reply:
`/tmp/kibana/node/v$nodeVersion`?

Reply:
I kind of like how this isn't global, it doesn't make any changes to the system that might conflict with other installs, but maybe that's irrelevant